homeassistant/components/dynalite/__init__.py

```diff
@@ -63,15 +78,71 @@ def num_string(value):
 
 PRESET_SCHEMA = vol.Schema({num_string: vol.Any(PRESET_DATA_SCHEMA, None)})
 
+TEMPLATE_ROOM_SCHEMA = vol.Schema(
+    {vol.Optional(CONF_ROOM_ON): num_string, vol.Optional(CONF_ROOM_OFF): num_string}
+)
 
-AREA_DATA_SCHEMA = vol.Schema(
+TEMPLATE_TRIGGER_SCHEMA = vol.Schema({vol.Optional(CONF_TRIGGER): num_string})
+
+TEMPLATE_TIMECOVER_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_NAME): cv.string,
-        vol.Optional(CONF_FADE): vol.Coerce(float),
-        vol.Optional(CONF_NO_DEFAULT): vol.Coerce(bool),
-        vol.Optional(CONF_CHANNEL): CHANNEL_SCHEMA,
-        vol.Optional(CONF_PRESET): PRESET_SCHEMA,
-    },
+        vol.Optional(CONF_CHANNEL_COVER): num_string,
+        vol.Optional(CONF_DEVICE_CLASS): cv.string,
+        vol.Optional(CONF_OPEN_PRESET): num_string,
+        vol.Optional(CONF_CLOSE_PRESET): num_string,
+        vol.Optional(CONF_STOP_PRESET): num_string,
+        vol.Optional(CONF_DURATION): vol.Coerce(float),
+        vol.Optional(CONF_TILT_TIME): vol.Coerce(float),
+    }
+)
+
+TEMPLATE_DATA_SCHEMA = vol.Any(
+    TEMPLATE_ROOM_SCHEMA, TEMPLATE_TRIGGER_SCHEMA, TEMPLATE_TIMECOVER_SCHEMA
+)
+
+TEMPLATE_SCHEMA = vol.Schema({str: TEMPLATE_DATA_SCHEMA})
```

Comment:
The templates don't seem to be the same thing as the cover platform. Please remove this change from this PR.

What does the templates do?

Reply:
there are some devices that are using multiple dynalite presets / channels

the cover is the one that absolutely need it, since each cover is usually done as three presets (open, close, stop) and one channel that some controls use to open and close (theoretically also to get the cover position, but it is not well supported). That is why I need to tell the library that these presets are connected together to form a cover device
The other two templates are ones i use a lot, but they can be removed and added in a separate PR.
template: room - usually each area in dynalite has multiple channels corresponding to the different lights in a room and two presets (1 for on and 4 for off) that turn on or off all the lights. This creates a switch device for the room that is either "on" or "off", but each one sends a command to a different preset
template: trigger - this is for a specfic preset that can be turned on and turns itself automatically off. usually something like a door buzzer. Creates one switch from it.

room and trigger can be moved to a different PR, but they are super small, are useful, and don't create any change here other than in the SCHEMA. Let me know if you want me to take them off. the "timecover" template is required (and is the more complex one) for a cover, otherwise there is no way to tell the library that these presets and channel form a cover device together

Reply:
So the templates are not part of the dynalite api? Then we will probably not accept them. Our standard is to only model what the device exposes and not add extra features on top. Extra features are already available in Home Assistant via automations and templates etc.

Reply:
the timecover template is definitely part of the API. You can see in their document (https://www.dynalite.org/public-download/2947/bd40c4247432c0917b35dda8c1e3bf05, page 24) that their RS485 blind control has these presets and channel in an area that controls a blind. The devices don't mean anything by themselves and I cannot really find corresponding ways to even map "fake" devices, as the interaction between them is complicated. For example, a "stop_fade" command on the channel causes the cover to stop, so if one is detected on the network, we know that the cover stopped opening or closing. This is their API for covers that they control on the backend with RS485.

For the room, this is also in their API (they have presets that control several lights, page 8 of the same document), and although it can be configured, to the best of my knowledge, even the defaults never get changed. This is just the way that Dynalite defines a switch. While this one may be doable with automations and templates, this is pretty much the standard for every installation of Dynalite and will complicate the configuration significantly.

trigger - this is not part of the api and i will remove it. the same behavior can be achieved by defining it as a preset switch, which is already existing in the functionality

Reply:
BTW, the Somfy blind control is different from the one in the integrations. The Somfy integration is for the cloud controlled blinds, while this is for the controllers that are connected by RS485 behind the gateway and don't have any direct IP connection to them

Reply:
removed the trigger template in the latest commit

Reply:
Ok, thanks for clarifying what templates mean here.

Reply:
sure. not sure why phillips decided to make it complicated...